mlir/lib/Dialect/Vector/Transforms/LowerVectorGather.cpp

```diff
@@ -96,6 +96,87 @@ struct FlattenGather : OpRewritePattern<vector::GatherOp> {
   }
 };
 
+/// Rewrites a vector.gather of a strided MemRef as a gather of a non-strided
+/// MemRef with updated indices that model the strided access.
+///
+/// ```mlir
+/// %subview = memref.subview %M (...) memref<100x3xf32> to memref<100xf32,
+/// strided<[3]>> %gather = vector.gather %subview (...) : memref<100xf32,
+/// strided<[3]>>
+/// ```
+/// ==>
+/// ```mlir
+/// %collapse_shape = memref.collapse_shape %M (...) memref<100x3xf32> into
+/// memref<300xf32> %1 = arith.muli %idxs, %c3 : vector<4xindex> %gather =
+/// vector.gather %collapse_shape (...) : memref<300xf32> (...)
+/// ```
```

Comment:
can you format the MLIR snippet a bit? Perhaps we can have one operation per line, e.g.,

```suggestion
/// ```mlir
/// %subview = memref.subview %M (...) memref<100x3xf32> to memref<100xf32,
///     strided<[3]>>
/// %gather = vector.gather %subview (...) : memref<100xf32, strided<[3]>>
/// ```
/// ==>
/// ```mlir
/// %collapse_shape = memref.collapse_shape %M (...) memref<100x3xf32> into
///     memref<300xf32>
/// %1 = arith.muli %idxs, %c3 : vector<4xindex>
/// %gather = vector.gather %collapse_shape (...) : memref<300xf32> (...)
/// ```
```

Reply:
Argh, that's clang-format, sorry :( Thanks for pointing it out! 🙏🏻